### PR TITLE
[Compat]: BetterEnd compatibility

### PR DESCRIPTION
**Pull request by JorisDeBeer**
All thanks to our new buddy in the Discord.
Balancing is a work in progress but its so big I checked it through didnt seem anything off. I will probably play with this mod anyway soon so should be fine

### CODE REVIEW

File: content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/eternal_crystal.json [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  "type": "MAKESHIFT",
+  "name": "eternal_crystal",
+  "parent": "makeshift_base",
+  "container": {
+    "type": "HOST",
+    "id": "betterend:eternal_crystal"
+  },
+  "properties": {
+    "grouped_attributes": [
+      {
```
Comment (SigmundGranaas) [inline content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/eternal_crystal.json:11]: Composite stats for a makeshift material is not needed. No tools are created.

File: content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/gelatine.json [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  "name": "gelatine",
+  "parent": "goo_base",
+  "container": {
+    "type": "HOST",
+    "id": "betterend:gelatine"
+  },
+  "properties": {
+    "grouped_attributes": [
+      {
+        "context": "COMPOSITE",
```
Comment (SigmundGranaas) [inline content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/gelatine.json:12]: Composite stats are not needed for this.

File: content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/lumecorn_rod.json [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  "name": "lumecorn_rod",
+  "parent": "scale_base",
+  "container": {
+    "type": "HOST",
+    "id": "betterend:lumecorn_rod"
+  },
+  "properties": {
+    "grouped_attributes": [
+      {
+        "context": "COMPOSITE",
```
Comment (SigmundGranaas) [inline content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/lumecorn_rod.json:12]: No composite stats needed here.

File: content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/raw_chorus_mushroom.json [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  "name": "raw_chorus_mushroom",
+  "parent": "hard_plant_base",
+  "container": {
+    "type": "HOST",
+    "id": "betterend:chorus_mushroom_raw"
+  },
+  "properties": {
+    "grouped_attributes": [
+      {
+        "context": "COMPOSITE",
```
Comment (SigmundGranaas) [inline content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/raw_chorus_mushroom.json:12]: No composite stats needed.

File: content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/silk_fiber.json [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  "parent": "string_base",
+  "resource_type": "DEFAULT",
+  "container": {
+    "type": "HOST",
+    "id": "betterend:silk_fiber"
+  },
+  "properties": {
+    "grouped_attributes": [
+      {
+        "context": "COMPOSITE",
```
Comment (SigmundGranaas) [inline content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/silk_fiber.json:13]: No composite stats needed.

File: content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/silk_moth_matrix.json [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  "name": "silk_moth_matrix",
+  "parent": "goo_base",
+  "container": {
+    "type": "HOST",
+    "id": "betterend:silk_moth_matrix"
+  },
+  "properties": {
+    "grouped_attributes": [
+      {
+        "context": "COMPOSITE",
```
Comment (SigmundGranaas) [inline content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/silk_moth_matrix.json:12]: No composite stats needed.

File: content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/small_amaranita_mushroom.json [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  "name": "small_amaranita_mushroom",
+  "parent": "hard_plant_base",
+  "container": {
+    "type": "HOST",
+    "id": "betterend:small_amaranita_mushroom"
+  },
+  "properties": {
+    "grouped_attributes": [
+      {
+        "context": "COMPOSITE",
```
Comment (SigmundGranaas) [inline content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/small_amaranita_mushroom.json:12]: No composite attributes needed

File: content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/small_jellyshroom.json [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+  "name": "small_jellyshroom",
+  "parent": "hard_plant_base",
+  "container": {
+    "type": "HOST",
+    "id": "betterend:small_jellyshroom"
+  },
+  "properties": {
+    "grouped_attributes": [
+      {
+        "context": "COMPOSITE",
```
Comment (SigmundGranaas) [inline content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/small_jellyshroom.json:12]: No composite stats needed.

File: content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/smaragdant_crystal_shard.json [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+  "name": "smaragdant_crystal_shard",
+  "parent": "crystal_base",
+  "container": {
+    "type": "HOST",
+    "id": "betterend:smaragdant_crystal_shard"
+  },
+  "properties": {
+    "grouped_attributes": [
+      {
+        "context": "COMPOSITE",
```
Comment (SigmundGranaas) [inline content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/smaragdant_crystal_shard.json:12]: No composite stats needed.

File: content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/sulphur_crystal.json [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+  "name": "sulphur_crystal",
+  "parent": "crystal_base",
+  "container": {
+    "type": "HOST",
+    "id": "betterend:sulphur_crystal"
+  },
+  "properties": {
+    "grouped_attributes": [
+      {
+        "context": "COMPOSITE",
```
Comment (SigmundGranaas) [inline content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/sulphur_crystal.json:12]: No composite stats needed.

File: content/forgero-compat/src/main/resources/assets/forgero/lang/en_us.json [hunk 11]
```diff
@@ -153,5 +153,53 @@
   "item.forgero.sythian": "Sythian",
   "item.forgero.white_mangrove": "White Mangrove",
   "item.forgero.witch_hazel": "Witch Hazel",
-  "item.forgero.zelkova": "Zelkova"
+  "item.forgero.zelkova": "Zelkova",
+  "item.forgero.aurora_crystal_shards": "Aurora Crystal Shards",
+  "item.forgero.ender_shard": "Ender Shard",
+  "item.forgero.eternal_crystal": "Eternal Crystal",
+  "item.forgero.smaragdant_crystal_shard": "Smaragdant Crystal Shard",
+  "item.forgero.sulphur_crystal": "Sulphur Crystal",
```
Comment (SigmundGranaas) [inline content/forgero-compat/src/main/resources/assets/forgero/lang/en_us.json]: This is called `Sulfur` in the mod, event though the item tag uses sulphur. I think we should stick to the same as the mod.

image

File: content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/umbralith.json [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+          },
+          {
+            "id": "stone-durability-offensive-upgrade",
+            "type": "DURABILITY",
+            "operation": "ADDITION",
+            "category": "OFFENSIVE",
+            "value": 95
+          },
+          {
+            "id": "stone-durability-utility-upgrade",
```
Comment (SigmundGranaas) [inline content/forgero-compat/src/main/resources/data/forgero/packs/betterend/material/umbralith.json]: Stones don't have any utility slots that they can be used in. As highlited by this PR: https://github.com/SigmundGranaas/forgero/pull/961

File: fabric/build.gradle [hunk 13]
```diff
@@ -117,8 +117,8 @@ curseforge {
 				optionalDependency "modmenu"
 				optionalDependency "mythicmetals"
 				optionalDependency "betternether"
+				optionalDependency "betterend"
 				optionalDependency "mystical-crops"
-				optionalDependency "mythicmetals"
```
Comment (SigmundGranaas) [inline fabric/build.gradle:121]: Why is this removed?

File: fabric/build.gradle [hunk 14]
```diff
@@ -117,8 +117,8 @@ curseforge {
 				optionalDependency "modmenu"
 				optionalDependency "mythicmetals"
 				optionalDependency "betternether"
+				optionalDependency "betterend"
```
Comment (SigmundGranaas) [inline fabric/build.gradle:120]: Would be nice to also add this to the modrinth block.

**Review by SigmundGranaas — CHANGES_REQUESTED** [review]
Awesome work! This is really well done. All the comments are just smaller things that should be fixed, but it is not critical.

The comments are just about removing attributes that cannot be applied in any configuration at the moment. I have probably missed some attributes, so it would probably be nice to go through again and see if we missed any.

Other that that, this seems good to go!

File: fabric/build.gradle (hunk 13 above)
Comment (JorisDeBeer) [reply fabric/build.gradle:121]: Its there twice look up :)

Comment (SigmundGranaas) [reply fabric/build.gradle:121]: I'm stupid

**Review by SigmundGranaas — APPROVED** [review]
Looks good! :star:
